src/Application.vala

```diff
@@ -175,6 +177,51 @@ public class Music.Application : Gtk.Application {
         playback_manager.shuffle ();
     }
 
+    private void action_open () {
+        var all_files_filter = new Gtk.FileFilter () {
+            name = _("All files"),
+        };
+        all_files_filter.add_pattern ("*");
+        var music_files_filter = new Gtk.FileFilter () {
+            name = _("Music Files"),
+        };
+        music_files_filter.add_mime_type ("audio/*");
+
+        var filter_model = new ListStore (typeof (Gtk.FileFilter));
+        filter_model.append (all_files_filter);
+        filter_model.append (music_files_filter);
+
+        var file_chooser = new Gtk.FileChooserNative (
```

Comment:
Gtk.FileChooserNative is deprecated since 4.10, so it would be better to avoid to use in new codes.